plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/TrinoS3ClientFactory.java

```diff
@@ -80,96 +82,112 @@ public class TrinoS3ClientFactory
     private final boolean enabled;
     private final int defaultMaxConnections;
 
+    private S3SelectStats stats;
+
     @GuardedBy("this")
-    private AmazonS3 s3Client;
+    private S3AsyncClient s3Client;
 
     @Inject
-    public TrinoS3ClientFactory(HiveConfig config)
+    public TrinoS3ClientFactory(HiveConfig config, S3SelectStats stats)
     {
         this.enabled = config.isS3SelectPushdownEnabled();
         this.defaultMaxConnections = config.getS3SelectPushdownMaxConnections();
+        this.stats = stats;
     }
 
-    synchronized AmazonS3 getS3Client(Configuration config)
+    synchronized S3AsyncClient getS3Client(Configuration config)
     {
         if (s3Client == null) {
             s3Client = createS3Client(config);
         }
         return s3Client;
     }
 
-    private AmazonS3 createS3Client(Configuration config)
+    private ClientOverrideConfiguration getClientOverrideConfig(RetryPolicy retryPolicy, String userAgentPrefix)
+    {
+        return ClientOverrideConfiguration.builder()
+                .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, userAgentPrefix)
+                .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_SUFFIX, enabled ? "Trino-select" : "Trino")
```

Comment:
I think this code isn't needed and never was, since this client factory is only for S3 Select. If S3 Select is not enabled, then this factory won't be used.

Reply:
makes sense, will remove this code.